### ISSUE TITLE
End date of Scenario.simulate() does not match the end date of a phase

### ISSUE THREAD

**Issue by lisphilar** [lisphilar/covid19-sir#96]
Summary:
End date of `Scenario.simulate()` does not match the end date of a phase.

CovsirPhy version 2.4.2

Related classes:
- `covsirphy.Scenario`

Codes and outputs:
```Python
import covsirphy as cs
# Dataset preparation
data_loader = cs.DataLoader("input")
jhu_data = data_loader.jhu()
population_data = data_loader.population()
scenario = cs.Scenario(jhu_data, population_data, country="Italy")
scenario.trend()
scenario.estimate(cs.SIRF)
scenario.add_phase(end_date="01Jan2020")
scenario.summary()
scenario.simulate()
```
The last date of summary was 01Jan2020, but the last date of simulated records was 22Dec2020.

Environment:
Python 3.8, pipenv, WSL.

**Comment by lisphilar** [comment]
This issue was caused by an error of calculation of the number of steps in simulation.
`end_date - start_date` was used.
`(next date of end_date) - start_date` must be used.

**Comment by lisphilar** [comment]
This will be fixed in version 2.5
